### PR TITLE
fix(TDP-5696): List preparations performance issue

### PR DESCRIPTION
**Pull request by fabballe**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-5696

**Please check if the PR fulfills these requirements**
- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-backend-service/src/main/java/org/talend/dataprep/conversions/inject/OwnerInjection.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+import org.talend.dataprep.api.preparation.PreparationDTO;
+import org.talend.dataprep.api.share.Owner;
+import org.talend.dataprep.preparation.store.PersistentPreparation;
+import org.talend.dataprep.security.Security;
+
+@Component
```
Comment (fabballe) [inline dataprep-backend-service/src/main/java/org/talend/dataprep/conversions/inject/OwnerInjection.java]: why add @component and @Scope if you create it by configuration (see in `Injections.java`)

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/populate/PopulateDatasetStep.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
+//  ============================================================================
```
Comment (fabballe) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/populate/PopulateDatasetStep.java]: do not push this file because it is not related to new implementation of preparation list

File: dataprep-test-api/src/test/resources/features/populate/50data100prep30step.feature [hunk 3]
```diff
@@ -0,0 +1,8 @@
+@Unstable
```
Comment (fabballe) [inline dataprep-test-api/src/test/resources/features/populate/50data100prep30step.feature]: same here

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/json/PipelineTransformer.java [hunk 4]
```diff
@@ -77,6 +78,9 @@
     @Autowired
     private StepMetadataRepository preparationUpdater;
 
+    @Autowired(required = false)
+    private Tracer tracer;
```
Comment (fabballe) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/json/PipelineTransformer.java]: Add `Optional<Tracer>`

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/json/PipelineTransformer.java [hunk 5]
```diff
@@ -117,11 +121,13 @@ public ExecutableTransformer buildExecutable(DataSet input, Configuration config
 
             @Override
             public void execute() {
+                final Span span = tracer.createSpan("pipeline-" + configuration.getPreparationId());
```
Comment (fabballe) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/json/PipelineTransformer.java]: and check if tracer is present

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/json/PipelineTransformer.java [hunk 6]
```diff
@@ -117,11 +121,13 @@ public ExecutableTransformer buildExecutable(DataSet input, Configuration config
...
                 try {
                     LOGGER.debug("Before transformation: {}", pipeline);
                     pipeline.execute(input);
                 } finally {
                     LOGGER.debug("After transformation: {}", pipeline);
+                    tracer.close(span);
```
Comment (fabballe) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/json/PipelineTransformer.java]: same here

File: dataprep-api/src/main/java/org/talend/dataprep/api/service/PreparationAPI.java [hunk 7]
```diff
@@ -242,17 +286,39 @@ public void move(@PathVariable(value = "id") @ApiParam(name = "id", value = "Id
...
         }
 
-        final PreparationDetailsGet enrichPreparation = getCommand(PreparationDetailsGet.class, preparationId, stepId);
         try {
-            PreparationMessage preparationMessage = mapper.readerFor(PreparationMessage.class).readValue(enrichPreparation.execute());
-            return beanConversionService.convert(preparationMessage, EnrichedPreparation.class);
+            final PreparationDTO preparationSummary = getPreparationSummary(preparationId, stepId);
+            return beanConversionService.convert(preparationSummary, PreparationDetailsDTO.class, //
+                    (dto, details) -> {
+                        final List<Action> actions = getCommand(PreparationGetActions.class, dto.getId()).execute();
```
[the remark this answers is not in the document]
Comment (fhuaulme) [reply dataprep-api/src/main/java/org/talend/dataprep/api/service/PreparationAPI.java]: 👍

File: dataprep-backend-service/src/main/java/org/talend/dataprep/configuration/DataPrepEvents.java [hunk 8]
```diff
@@ -45,4 +46,9 @@ public ApplicationEventMulticaster getDataPrepEventsCaster( ) {
         return new DataPrepEventsCaster(executor, beanFactory);
     }
 
+    @Bean
```
[the remark this answers is not in the document]
Comment (fhuaulme) [reply dataprep-backend-service/src/main/java/org/talend/dataprep/configuration/DataPrepEvents.java]: It was a way to always push traces to Zipkin and not rely on probabilistic push. Removed.

File: dataprep-backend-service/src/main/java/org/talend/dataprep/conversions/inject/OwnerInjection.java (hunk 1 above)
Comment (fhuaulme) [reply dataprep-backend-service/src/main/java/org/talend/dataprep/conversions/inject/OwnerInjection.java]: indeed.

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/populate/PopulateDatasetStep.java (hunk 2 above)
Comment (fhuaulme) [reply dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/populate/PopulateDatasetStep.java]: 👍

File: dataprep-test-api/src/test/resources/features/populate/50data100prep30step.feature (hunk 3 above)
Comment (fhuaulme) [reply dataprep-test-api/src/test/resources/features/populate/50data100prep30step.feature]: 👍

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/json/PipelineTransformer.java (hunk 4 above)
Comment (fhuaulme) [reply dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/json/PipelineTransformer.java]: 👍

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/json/PipelineTransformer.java (hunk 5 above)
Comment (fhuaulme) [reply dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/json/PipelineTransformer.java]: 👍

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/json/PipelineTransformer.java (hunk 6 above)
Comment (fhuaulme) [reply dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/json/PipelineTransformer.java]: 👍
